shared/chat/conversation/info-panel/container.js

```diff
@@ -21,15 +21,18 @@ type OwnProps = {
 const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
   const conversationIDKey = ownProps.conversationIDKey
   const meta = Constants.getMeta(state, conversationIDKey)
+  const lastMessage = Constants.getLastMessage(state, conversationIDKey)
 
   let admin = false
   let canEditChannel = false
   let canSetRetention = false
+  let canDeleteHistory = false
   if (meta.teamname) {
     const yourOperations = getCanPerform(state, meta.teamname)
     admin = yourOperations.manageMembers
     canEditChannel = yourOperations.editChannelDescription
     canSetRetention = yourOperations.setRetentionPolicy
+    canDeleteHistory = lastMessage && (meta.teamType === 'adhoc' || yourOperations.deleteChatHistory)
```

Comment:
why isn't yourOperations just have this as true for adhoc, why do we need these extra checks?

Reply:
I don't know, I just copied the existing logic.

Reply:
`teamType` seems to be a conversation property, not a team property. Maybe we need a `chat1.canUserPerform()`?